src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.PerAnalyzerState.cs

```diff
@@ -23,6 +23,7 @@ private class PerAnalyzerState
 
             private Dictionary<SyntaxTree, AnalyzerStateData> _lazySyntaxTreesWithAnalysisData = null;
             private int _pendingSyntaxAnalysisTreesCount = 0;
+            private Dictionary<ISymbol, AnalyzerStateData> _lazyPendingSymbolEndAnalyses = null;
 
```

Comment:
A given analyzer is only permitted to have *one* symbol end action? That is, it cannot register, for example, two different end actions and expect them both to occur? If that is true, it should be documented.

Reply:
No, an analyzer can have multiple symbol end actions registered for every symbol. [This code ](https://github.com/dotnet/roslyn/pull/28918/files#diff-3fd5f74b371dcbb1f98fc79be65b56fdR468) executes all the relevant symbol end actions for an analyzers for a specific symbol.